File: examples/javascript/streaming.mjs
/**
 * Streaming Example
 *
 * Demonstrates streaming chat completions using the async iterator pattern.
 *
 * Run: node examples/javascript/streaming.mjs
 *
 * Requires LITEFORGE_API_KEY environment variable.
 */

import {
  AsyncForgeClient,
  createMessageSystem,
  createMessageUser,
} from '@forge/sdk';

const client = new AsyncForgeClient();

const messages = [
  createMessageSystem('You are a creative storyteller.'),
  createMessageUser('Tell me a very short story about a robot learning to paint.'),
];

console.log('Streaming response:\n');

const stream = await client.completeStream(messages);

let chunk;
while ((chunk = await stream.next()) !== null) {
  const content = chunk.choices[0]?.delta?.content;
  if (content) {
    process.stdout.write(content);
  }
}

console.log('\n\nStream complete.');

File: crates/liteforge/examples/streaming.rs
//! Streaming completion example.
//!
//! Run with: cargo run --example streaming

use futures::StreamExt;
use liteforge::{AsyncForgeClient, Message};

#[tokio::main]
async fn main() {
    // Create async client
    let client = AsyncForgeClient::new();

    println!("Using model: {}", client.model());

    // Create a streaming completion
    let messages = vec![Message::user("Tell me a short story about a robot.")];

    match client.complete_stream(messages).await {
        Ok(mut stream) => {
            print!("\nResponse: ");
            while let Some(result) = stream.next().await {
                match result {
                    Ok(chunk) => {
                        if let Some(content) = chunk.content() {
                            print!("{}", content);
                        }
                    }
                    Err(e) => {
                        eprintln!("\nStream error: {}", e);
                        break;
                    }
                }
            }
            println!();
        }
        Err(e) => {
            eprintln!("Error: {}", e);
        }
    }
}
